cpg_pipes/providers/inputs.py

```diff
@@ -127,12 +131,21 @@ def populate_analysis(self, cohort: Cohort) -> None:
 
     @abstractmethod
     def populate_pedigree(
-        self, 
+        self,
         cohort: Cohort,
         ped_files: list[Path] | None = None,
     ) -> None:
         """
-        Populate pedigree data
+        Populate pedigree data from file
+        """
+
+    @abstractmethod
+    def populate_pedigree_from_db(
```

Comment:
The abstract class shouldn't know about the DB. DB is a detail of the SMDB implementation, and in contrast, the CSV implementation wouldn't know what to do in the `populate_pedigree_from_db` method vs. standard `populate_pedigree`.

I think there should be only one method called `populate_pedigree`.

Reply:
Good point! I've combined the two now.

Reply:
Thanks! We might even want to move the ped parsing block into the abstract class, because I imagine it should be able to parse them - if provided - for any provider implementation.